staging/src/k8s.io/apiserver/pkg/endpoints/filters/audit.go

```diff
@@ -230,14 +226,14 @@ func (a *auditResponseWriter) Unwrap() http.ResponseWriter {
 
 func (a *auditResponseWriter) processCode(code int) {
 	a.once.Do(func() {
-		if a.event.ResponseStatus == nil {
-			a.event.ResponseStatus = &metav1.Status{}
+		ac := audit.AuditContextFrom(a.ctx)
+		if ac.GetEventResponseStatus() == nil {
+			ac.SetEventResponseStatus(&metav1.Status{})
 		}
-		a.event.ResponseStatus.Code = int32(code)
-		a.event.Stage = auditinternal.StageResponseStarted
+		ac.GetEventResponseStatus().Code = int32(code)
```

Comment:
this does not seem right

Reply:
Any suggestions to fix the concern?

Reply:
Hmmm, I see, this is different from the other case because we may already have a `ResponseStatus` set. I missed that originally. Let's keep the current then (perhaps with Tim's suggested improvement).

Reply:
sounds good.